packages/@lwc/engine-core/src/framework/decorators/track.ts

```diff
@@ -19,12 +19,16 @@ import { logError } from '../../shared/logger';
  * LWC Components. This function can also be invoked directly
  * with any value to obtain the trackable version of the value.
  */
-export default function track(
-    value: unknown,
-    context: ClassMemberDecoratorContext | string | symbol
+export default function track<Class extends LightningElement>(
+    target: undefined,
+    context: ClassFieldDecoratorContext<Class>
 ): void;
-export default function track<T>(target: T): T {
-    if (arguments.length === 1) {
+export default function track<T>(target: T, context?: never): T;
+export default function track(
+    target: unknown,
+    context?: ClassFieldDecoratorContext | undefined
+): unknown {
+    if (isUndefined(context)) {
```

Comment:
This is technically an observable change. At first I thought this function might be unused, due to our Karma tests:

![Screenshot 2024-08-02 at 2 37 07 PM](https://github.com/user-attachments/assets/171738e3-13d3-4dad-98e1-94aac76ecced)

And looking at the sibling `api.ts` and `wire.ts` files, those functions merely throw an error. So you'd think we _could_ do that here, but then that would break this case:

```js
import { track } from 'lwc'

const proxy = track(object)
```

I tested if this is possible on-platform, and it's not:

```
 "track" can only be used as a class decorator
```

... which is due to this rule:

https://github.com/salesforce/lwc/blob/9e4ea5985cf15b7eda2b5b4a66c8c780120bfbfc/packages/%40lwc/errors/src/compiler/error-info/lwc-class.ts#L111-L116

However, this API may be used internally. So to remove it, we would probably want to start with [reporting](https://github.com/salesforce/lwc/blob/9e4ea5985cf15b7eda2b5b4a66c8c780120bfbfc/packages/%40lwc/engine-core/src/framework/reporting.ts) to see who might be using it.

So in short, the observable change here is this:

```js
import { track } from 'lwc'

// used to fail, will now pass
const proxy = track(object, undefined)
```

This is probably fine, but just calling it out.

Reply:
Elaborate workarounds: https://github.com/salesforce/lwc/pull/4434

Reply:
We also get a type error for `track(object, undefined, anything)`, but not a runtime error. I forget why I changed this, but it's not important, so I think I'll just change it back.